feature/experimental/policy/otg_tests/prefix_set_test/prefix_set_test.go

```diff
@@ -100,45 +108,47 @@ func TestPrefixSet(t *testing.T) {
 
 func TestPrefixSetWithCLIConfig(t *testing.T) {
 	dut := ondatra.DUT(t, "dut")
-	ctx := context.Background()
-	cli := dut.RawAPIs().CLI(t)
-
-	switch dut.Vendor() {
-	case ondatra.ARISTA:
-		helpers.GnmiCLIConfig(t, dut, `
-			ip prefix-list TAG_3_IPV4
-				seq 10 permit 10.240.31.48/28
-				seq 20 permit 10.244.187.32/28
-				seq 30 permit 173.36.128.0/20
-				seq 40 permit 173.37.128.0/20
-				seq 50 permit 173.38.128.0/20
-				seq 60 permit 173.39.128.0/20
-				seq 70 permit 173.40.128.0/20
-				seq 80 permit 173.41.128.0/20
-				seq 90 permit 173.42.128.0/20
-				seq 100 permit 173.43.128.0/20
-			`,
-		)
-		helpers.GnmiCLIConfig(t, dut, `
-			management api gnmi
-				transport grpc default
-				operation set persistence
-			`,
-		)
-		cmd := "agent Octa terminate"
-		res, err := cli.RunCommand(ctx, "agent Octa terminate")
-		if err != nil {
-			t.Errorf("error executing command %q:\n%v", cmd, err)
-		}
-		if res.Error() != "" {
-			t.Errorf("error executing command %q:\n%v", cmd, res.Error())
-		}
-	}
 
 	dutOcRoot := &oc.Root{}
 	rp := dutOcRoot.GetOrCreateRoutingPolicy()
 	ds := rp.GetOrCreateDefinedSets()
-	v4PrefixSet := ds.GetOrCreatePrefixSet("TAG_3_IPV4")
+	v4PrefixSet := ds.GetOrCreatePrefixSet(tag3IPv4)
+	v4PrefixSet.GetOrCreatePrefix("10.240.31.48/28", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("10.244.187.32/28", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.36.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.37.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.38.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.39.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.40.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.41.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.42.128.0/20", mskLen)
+	v4PrefixSet.GetOrCreatePrefix("173.43.128.0/20", mskLen)
+	gnmi.Update(t, dut, gnmi.OC().RoutingPolicy().DefinedSets().PrefixSet(tag3IPv4).Config(), v4PrefixSet)
+	prefixSet := gnmi.Get[*oc.RoutingPolicy_DefinedSets_PrefixSet](t, dut, gnmi.OC().RoutingPolicy().DefinedSets().PrefixSet(tag3IPv4).State())
+	if len(prefixSet.Prefix) != 10 {
+		t.Errorf("Prefix set has %v prefixes, want 10", len(prefixSet.Prefix))
+	}
+
+	gnoiClient := dut.RawAPIs().GNOI(t)
+	if _, ok := processName[dut.Vendor()]; !ok {
+		t.Fatalf("Please add support for vendor %v in var processName", dut.Vendor())
+	}
+	killProcessRequest := &spb.KillProcessRequest{
+		Signal:  spb.KillProcessRequest_SIGNAL_TERM,
+		Name:    processName[dut.Vendor()],
+		Pid:     findProcessByName(context.Background(), t, dut, processName[dut.Vendor()]),
+		Restart: true,
+	}
+	processKillResponse, err := gnoiClient.System().KillProcess(context.Background(), killProcessRequest)
+	if err != nil {
+		t.Fatalf("Failed to restart process %v with unexpected err: %v", processName[dut.Vendor()], err)
+	}
```

Comment:
please use a device os primitive to restart the process. the gnoi, gnmi services may run within the same process depending on the implementation, so there's no way to guarantee getting a successful response.

Reply:
By device os primitive do you mean a CLI command?

I see the issue you are referring to though.  If a process is killed which hosts the gnoi service, no response is expected.  This code could explicitly define this dependency.   ie: if for example `octa` hosts gnoi and gnmi, then we can expect no response to the gnoi and no ability to check /system/processes for some period of time.  In this scenario we can still do a gnmi subscribe to find the processes and retry it a few times until some timeout.

thoughts?

Reply:
> By device os primitive do you mean a CLI command?

either vendor cli (for arista, this will be `agent Octa terminate`) or the underlying OS command, for linux `sudo kill -SIGKILL <pid>`

> In this scenario we can still do a gnmi subscribe to find the processes and retry it a few times until some timeout.

yeah, this sounds reasonable.

Reply:
using `agent Octa terminate` now